### ISSUE TITLE
db hint

### ISSUE THREAD

**Issue by vitaly-t** [dropTHATdatabase/project3#32]
about such code:
```js
'use strict';

const path = require('path');
const db = require(path.join(__dirname, 'index.js'));

// Create the users table and exit application
// Only intended to be run from the console
db.users.create()
  .then(() => {
    db.hunts.create()
      .then(() => {
        console.log('hunts table created');
        db.clues.create()
          .then(() => {
            console.log('clues table created');
            db.participants.create()
              .then(() => {
                console.log('participants table created');
                db.clues_users.create()
                  .then(() => {
                    console.log('clues_users table created');
                    process.exit();
                  }).catch(console.log);
              }).catch(console.log);
          }).catch(console.log);
      }).catch(console.log);
    console.log('users table created');
  }).catch(console.log);
```
it is more logical to do this as a plain list inside a transaction, like this:

```js
db.tx(t=>t.batch([
        t.users.create, t.clues.create, t.participants.create, t.clues.create
    ]))
    .then(data=> {
        console.log('all created');
    })
    .catch(error=> {
        // all rolled back due to an error;
    });
```
you you do not need to use `process.exit();`. Instead, call `pgp.end()` in the end, if you want to kill the connection pool and exit right away. See [Library de-initialization](https://github.com/vitaly-t/pg-promise#library-de-initialization).

Also, interesting, you are using the protocol extended, but I couldn't find where you are extending it...

Ops, I got it, you mixed up the notion of `db`, it is not what is suggested within [pg-promise-demo](https://github.com/vitaly-t/pg-promise-demo). You will find that precise pattern would result in a more logical and ordered code, via repositories.

**Comment by harryganz** [comment]
So I am not actually extending pg, which is why I didn't have access to the tx function to make transactions. I looked at the demo and that does seem like a better way to do it, thanks.

Also, how do you find all the people using pg-promise?

**Comment by vitaly-t** [comment]
>Also, how do you find all the people using pg-promise?

GitHub search engine is good, you can see all the new references to your library.

**Comment by harryganz** [comment]
I tried doing a code seach in the github api and it said that I needed to give it a user or organization. The same limitation doesn't seem to apply to their search feature. Is there a way to do a code search using the api?

**Comment by vitaly-t** [comment]
I've only used the UI they provide, not sure about the internal API.

**Comment by harryganz** [comment]
Closed #34
